doc/commands.md

```diff
@@ -202,6 +202,26 @@ arguments on the left side of the `≔` symbol (similarly to a value declaration
 in OCaml). Some arguments can be declared as implicit by enclosing them in
 curly brackets.
 
+<!---------------------------------------------------------------------------->
+### `inductive`
+The command `inductive` is used to define an inductive type.
```

Comment:
The command `inductive` can be used to define an inductive type, its constructors and its associated induction principle if it can be generated. For the moment, we only support first-order data types.

Reply:
Yes!
Why the name  of induction principle isn't correct if the name of inductive type is \mathbb{N} ?

Reply:
Good point. This is because ℕ_ind is not a declared identifier. If the name of the type is not a regular identifier, then we should declare the name of the induction principle using Sign.add_ident.

Reply:
How can I do this test?

Reply:
@fblanqui , how I can check if the name of a type is a regular identifier or not?

Reply:
Unless someone knows if we can reuse Parser.regular_ident, we can use http://caml.inria.fr/pub/docs/manual-ocaml/libstr.html . @elhaddadyacine recently had the same problem with ZenonModulo. So, it would be good to share a function doing that in LP.

Reply:
I don't know exactly the problem of @elhaddadyacine but in my case, I can just test if the name of a type (or any string) belongs in signature.sign_idents, like this:
`if StrSet.mem id.elt !(ss.signature.sign_idents) then`. Right?

Reply:
Good idea but this is not sufficient as it doesn't tell you whether your identifier is escaped, that is, of the form `{|...|}` (check that the first character is `{`). In this case, you should call it `{|ind_...|}`. We could do the same for declared idents.

Reply:
Ask Yacine to know what he did.

Reply:
See https://github.com/elhaddadyacine/ekstrakto/commit/5a9557e19823b5a989d3f54f5c163bb66f416ada

Reply:
Ok but 1) this does not use dune and LP relies on dune 2) it is not necessary anymore as, finally, Str is not needed in your code. (FYI it is used in lsp but could disappear in the next PR by Emilio and François.)